skfda/clustering.py

```diff
@@ -0,0 +1,1180 @@
+"""Clustering Module."""
+
+import numpy as np
+from .grid import FDataGrid
+from .metrics import pairwise_distance, lp_distance, norm_lp
+import matplotlib.pyplot as plt
+from mpldatacursor import datacursor
+import matplotlib.patches as mpatches
+from matplotlib.ticker import MaxNLocator
+import random
+from abc import ABC, abstractmethod
+
+__author__ = "Amanda Hernando Bernabé"
+__email__ = "amanda.hernando@estudiante.uam.es"
+
+
+class ClusteringData(ABC):
```

Comment:
I would change the name, as this ABC is only for K-means and Fuzzy K-means, and not for other algorithms, such as Hierarchical clustering.

Reply:
The idea is that if other algorithms were implemented, they could inherit from this class too.

Reply:
But not all clustering algorithms need those arguments.

Reply:
Should `ClusteringData`  inherit from the sklearn classes `BaseEstimator`, `ClusterMixin` and `TransformerMixin`?

https://github.com/scikit-learn/scikit-learn/blob/7b136e92acf49d46251479b75c88cba632de1937/sklearn/cluster/k_means_.py#L771
https://github.com/scikit-learn/scikit-learn/blob/7b136e92acf49d46251479b75c88cba632de1937/sklearn/base.py

Reply:
I think so.